stock_move_propagate_first_move/models/stock_move.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    first_move_id = fields.Many2one(
+        comodel_name="stock.move",
+        string="First Move",
+        readonly=True,
+        help="The original move which generated this one.",
+    )
+    first_picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type",
+        string="Original Operation Type",
+        related="first_move_id.picking_type_id",
+        store=True,
+        help="Picking type of the original move which generated this one.",
+    )
```

Comment:
@sbejaoui  IMO we should always initialize the first_move_id with self if not provided at create. In this way it's not required to always check if first_move_id is set...

```suggestion
    )
    
    @api.model_create_multi
    def create(self, vals_list):
        res = super().create(vals_list)
        for res in res:
            if res.first_move_id:
                continue
            res.first_move_id = res
        return res
```